src/algebra/ring/basic.lean

```diff
@@ -1168,3 +1189,97 @@ theorem neg_left : commute a b → commute (- a) b := semiconj_by.neg_left
 @[simp] theorem sub_left : commute a c → commute b c → commute (a - b) c := semiconj_by.sub_left
 
 end commute
+section prop_typeclasses
+
+/-! ### Propositional typeclasses on several ring equivalences, useful in the theory of semilinear
+maps
+
+-/
+variables {R₁ : Type*} {R₂ : Type*} {R₃ : Type*}
+variables [semiring R₁] [semiring R₂] [semiring R₃]
```

Comment:
I think these probably make more sense in their own file, `ring/comp_typeclasses.lean`

Reply:
Good point -- done!